packages/cli/src/controller/migrate/manifest/migrate-manifest.controller.ts

```diff
@@ -0,0 +1,121 @@
+// Copyright 2020-2024 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: GPL-3.0
+
+import fs from 'fs';
+import path from 'path';
+import {NETWORK_FAMILY} from '@subql/common';
+import {CommonSubqueryProject} from '@subql/types-core';
+import {upperFirst} from 'lodash';
+import YAML from 'yaml';
+import {findRunnerByNetworkFamily, prepareDirPath, renderTemplate} from '../../../utils';
+import {TemplateKind} from '../../codegen-controller';
+import {graphToSubqlNetworkFamily, networkDsConverters, networkTemplateConverters} from '../constants';
+import {getChainIdByNetworkName} from '../migrate-controller';
+import {MigrateDatasourceKind, SubgraphDataSource, SubgraphProject, SubgraphTemplate, ChainInfo} from '../types';
+
+const PROJECT_TEMPLATE_PATH = path.resolve(__dirname, '../../../template/project.ts.ejs');
+
+export function readSubgraphManifest(inputPath: string): SubgraphProject {
+  if (!fs.existsSync(inputPath)) {
+    throw new Error(`Subgraph manifest ${inputPath} is not exist`);
+  }
+  const subgraphManifest: SubgraphProject = YAML.parse(fs.readFileSync(inputPath, 'utf8'));
+  return subgraphManifest;
+}
+
+export function extractNetworkFromManifest(subgraphProject: SubgraphProject): ChainInfo {
+  const firstDsKind = subgraphProject.dataSources[0].kind;
+  const firstDsNetwork = subgraphProject.dataSources[0].network;
+  if (!firstDsKind || !firstDsNetwork) {
+    throw new Error(`Subgraph dataSource kind or network not been found`);
+  }
+  const networkFamily = graphToSubqlNetworkFamily[firstDsKind];
+  if (!networkFamily) {
+    throw new Error(`Corresponding subquery network is not find with subgraph data source kind ${firstDsKind}`);
+  }
+  return {networkFamily, chainId: getChainIdByNetworkName(networkFamily, subgraphProject.dataSources[0].network)};
+}
+
+/**
+ * Render subquery project to .ts file
+ * @param projectPath
+ * @param project
+ */
+export async function renderManifest(projectPath: string, project: CommonSubqueryProject): Promise<void> {
+  try {
+    await renderTemplate(PROJECT_TEMPLATE_PATH, projectPath, {
+      props: {
+        importTypes: {
+          network: 'ethereum',
+          projectClass: 'EthereumProject',
+          projectDatasourceKind: 'EthereumDatasourceKind',
+          projectHandlerKind: 'EthereumHandlerKind',
+        },
+        projectJson: project,
+      },
+      helper: {
+        upperFirst,
+      },
+    });
+  } catch (e) {
```

Comment:
We need a way we can expose this underlying error. If a user faces an error here, we have no way to debug it without making a build. 

We could set a log level or include the same error